tools/commands/build.py

```diff
@@ -66,37 +90,93 @@ def make(self):
         logger.info("building & installing {}: {}".format(self.name, " ".join(cmd)))
         subprocess.check_call(cmd, env=self.env)
 
+class acbuilder(builder):
+    def __init__(self, name, modules_dir, make_verbose=False):
+        self.make_verbose = make_verbose
+        super(acbuilder, self).__init__(name, modules_dir)
+
+    def clean(self):
+        if os.path.exists(self.build_path):
+            os.system("make -C {} uninstall".format(self.build_path))
+        super(acbuilder, self).clean()
+
+    def prepare(self):
+        os.chdir(self.src_path)
+        cmd = ["./build-tools/autogen.sh"]
+        logger.info("preparing {}: {}".format(self.name, " ".join(cmd)))
+        subprocess.check_call(cmd, env=self.env)
+
+        self.configure()
+
+    def configure(self):
+        os.mkdir(self.build_path)
```

Comment:
This will error if you run it a second time. So:
```suggestion
        try:
            os.mkdir(self.build_path)
        except OSError:
            pass
```
(that way, if it failed and it didn't actually exist, the chdir below will still fail).